reframe/frontend/executors/policies.py

```diff
@@ -41,14 +42,15 @@ def _cleanup_all(tasks, *args, **kwargs):
 
 def _print_perf(task):
     '''Get performance info of the current task.'''
-
-    perfvars = task.testcase.check.perfvalues
-    for key, info in perfvars.items():
-        name = key.split(':')[-1]
-        getlogger().info(
-            f'P: {name}: {info[0]} {info[4]} '
-            f'(r:{info[1]}, l:{info[2]}, u:{info[3]})'
-        )
+    
+    with change_dir(task.testcase.check.stagedir or '.'):
```

Comment:
Which is the case that we might not have stagedir?

Reply:
It seems that it was the case for some unittests. Maybe because it was a temporary directory.

Reply:
Hm, I have a look and I think it's because we are calling `_print_perf` in every failure (even if the test fails early, before stagedir is set). I think it's better we only change directory if there are perfvars:
```python
    perfvars = task.testcase.check.perfvalues
    if perfvars:
        with change_dir(task.testcase.check.stagedir):
```
 If we have perfvars then the test should have run and have a stagedir, right?

Reply:
thanks @ekouts

Reply:
This `_print_perf()` is therefore a bug that I introduced with #2618 😬 What was your scenario that caused it to fail. It'd be good to add a unit test to check this.